File: projetos-fenix-para-estudo/t2ti-erp-fenix-menu/fontes/frontend/mvc/lib/src/model/pessoa_juridica.dart
class PessoaJuridica implements Comparable<PessoaJuridica> {
  PessoaJuridica(
    this._id,
    this._idPessoa,
    this._cnpj,
    this._nomeFantasia,
    this._inscricaoEstadual,
    this._inscricaoMunicipal,
    this._dataConstituicao,
    this._tipoRegime,
    this._crt,
  );

  String 
        _id,
        _idPessoa,
        _cnpj,
        _nomeFantasia,
        _inscricaoEstadual,
        _inscricaoMunicipal,
        _dataConstituicao,
        _tipoRegime,
        _crt;

  String get id => _id;
  set id(String value){
    if (value == null) value = "";
    _id = value;
  }

  String get idPessoa => _idPessoa;
  set idPessoa(String value){
    if (value == null) value = "";
    _idPessoa = value;
  }

  String get cnpj => _cnpj;
  set cnpj(String value){
    if (value == null) value = "";
    _cnpj = value;
  }

  String get nomeFantasia => _nomeFantasia;
  set nomeFantasia(String value){
    if (value == null) value = "";
    _nomeFantasia = value;
  }

  String get inscricaoEstadual => _inscricaoEstadual;
  set inscricaoEstadual(String value){
    if (value == null) value = "";
    _inscricaoEstadual = value;
  }

  String get inscricaoMunicipal => _inscricaoMunicipal;
  set inscricaoMunicipal(String value){
    if (value == null) value = "";
    _inscricaoMunicipal = value;
  }

  String get dataConstituicao => _dataConstituicao;
  set dataConstituicao(String value){
    if (value == null) value = "";
    _dataConstituicao = value;
  }

  String get tipoRegime => _tipoRegime;
  set tipoRegime(String value){
    if (value == null) value = "";
    _tipoRegime = value;
  }

  String get crt => _crt;
  set crt(String value){
    if (value == null) value = "";
    _crt = value;
  }

  PessoaJuridica.fromMap(Map m) {
    id = m["id"];
    idPessoa = m["idPessoa"];
    cnpj = m["cnpj"];
    nomeFantasia = m["nomeFantasia"];
    inscricaoEstadual = m["inscricaoEstadual"];
    inscricaoMunicipal = m["inscricaoMunicipal"];
    dataConstituicao = m["dataConstituicao"];
    tipoRegime = m["tipoRegime"];
    crt = m["crt"];
  }

  Map<String, dynamic> get toMap {
    return {
      "id":_id,
      "idPessoa":_idPessoa,
      "cnpj":_cnpj,
      "nomeFantasia":_nomeFantasia,
      "inscricaoEstadual":_inscricaoEstadual,
      "inscricaoMunicipal":_inscricaoMunicipal,
      "dataConstituicao":_dataConstituicao,
      "tipoRegime":_tipoRegime,
      "crt":_crt,
    };
  }  

  @override
  int compareTo(PessoaJuridica other) => _cnpj.compareTo(other._cnpj);
}

File: projetos-fenix-para-estudo/t2ti-erp-fenix-banco-agencia-pessoa-completos/fontes/backend/delphi/dmvc/Model/Cadastros/PessoaJuridica.pas
unit PessoaJuridica;

interface

uses
  MVCFramework.Serializer.Commons, ModelBase;

type

  [MVCNameCase(ncLowerCase)]
  TPessoaJuridica = class(TModelBase)
  private
    FID: Integer;
    FID_PESSOA: Integer;
    FCNPJ: string;
    FNOME_FANTASIA: string;
    FINSCRICAO_ESTADUAL: string;
    FINSCRICAO_MUNICIPAL: string;
    FDATA_CONSTITUICAO: TDateTime;
    FTIPO_REGIME: string;
    FCRT: string;
  public
    procedure ValidarInsercao; override;
    procedure ValidarAlteracao; override;
    procedure ValidarExclusao; override;

    [MVCColumnAttribute('ID', True)]
    [MVCNameAsAttribute('id')]
    property Id: Integer  read FID write FID;
    [MVCColumnAttribute('ID_PESSOA')]
    [MVCNameAsAttribute('idPessoa')]
    property IdPessoa: Integer  read FID_PESSOA write FID_PESSOA;
    [MVCColumnAttribute('CNPJ')]
    [MVCNameAsAttribute('cnpj')]
    property Cnpj: string  read FCNPJ write FCNPJ;
    [MVCColumnAttribute('NOME_FANTASIA')]
    [MVCNameAsAttribute('nomeFantasia')]
    property NomeFantasia: string  read FNOME_FANTASIA write FNOME_FANTASIA;
    [MVCColumnAttribute('INSCRICAO_ESTADUAL')]
    [MVCNameAsAttribute('inscricaoEstadual')]
    property InscricaoEstadual: string  read FINSCRICAO_ESTADUAL write FINSCRICAO_ESTADUAL;
    [MVCColumnAttribute('INSCRICAO_MUNICIPAL')]
    [MVCNameAsAttribute('inscricaoMunicipal')]
    property InscricaoMunicipal: string  read FINSCRICAO_MUNICIPAL write FINSCRICAO_MUNICIPAL;
    [MVCColumnAttribute('DATA_CONSTITUICAO')]
    [MVCNameAsAttribute('dataConstituicao')]
    property DataConstituicao: TDateTime  read FDATA_CONSTITUICAO write FDATA_CONSTITUICAO;
    [MVCColumnAttribute('TIPO_REGIME')]
    [MVCNameAsAttribute('tipoRegime')]
    property tipoRegime: string  read FTIPO_REGIME write FTIPO_REGIME;
    [MVCColumnAttribute('CRT')]
    [MVCNameAsAttribute('crt')]
    property Crt: string  read FCRT write FCRT;
  end;

implementation

{ TPessoaJuridica }

procedure TPessoaJuridica.ValidarInsercao;
begin
  inherited;

end;

procedure TPessoaJuridica.ValidarAlteracao;
begin
  inherited;

end;

procedure TPessoaJuridica.ValidarExclusao;
begin
  inherited;

end;

end.
